### ISSUE TITLE
Vertical scroll offset of removed element detected as 0

### ISSUE THREAD

**Issue by nickschot** [ember-animation/ember-animated#10]
I have a (removed) sprite which I animate with a custom Motion which behaves kind of like `toRight` from liquid-fire. If the page is scrolled down before the transition begins scrollY is still detected as 0. I suspect the detection happens "too late".

I have tried to track down why this happens, but have not yet succeeded. Within the `findOffsets` method in the Sprite class (where scroll detection happens) the scroll position is already 0. It is correctly detecting that the effective offset parent is `body`.

When manually checking window.scrollY before the transition happens, the scroll size is of course detected correctly.

There are no other scroll containers.

**Comment by nickschot** [comment]
Random thought: is the body scroll already reset because the actual transition is already happening?

**Comment by ef4** [comment]
Are you using an animated-container around your sprites? Because without it the document height may collapse when we pull the sprite out of static positioning.

**Comment by nickschot** [comment]
Yes I do. I'll try to create an isolated reproduction on twiddle in a bit.

**Comment by nickschot** [comment]
I had some trouble with ember-twiddle, so I have created a minimal reproduction on github: https://github.com/nickschot/ember-animated-scroll-reproduction .

Steps:
- Open app in responsive mode
- Go to not-index route
- Scroll down
- Press "go to index" button

**Comment by nickschot** [comment]
I am much further with this right now, problem is probably related to big elements being rendered in animated-orphans causing weird changes in the height of the viewport/html. I'll update here with a possible workaround/solution.

**Comment by sdhull** [comment]
@ef4 Not sure if this is related, but I noticed a similar bug from your (awesome!) [presentation](https://github.com/ef4/living-animation) slide 23 yesterday

If you scroll down on `/tutorial-23`, it breaks the animation that goes to the detail page (however animation going back to index mostly works still)

**Comment by ef4** [comment]
Yes, the slides totally don't do the right thing with body scroll right now. ember-animated needs a good built-in solution to compensating for the size of the body changing.

I think we are going to just need to treat the `<body>` as a special kind of `{{#animated-container}}`.

**Comment by sdhull** [comment]
Awesome! I wasn't sure I should even mention it as I figured you already were aware 😬

**Comment by chrism** [comment]
I'm also running into this issue with the height of the element collapsing during an animation.

Using an `{{#animated-container}}` introduces extra markup which breaks my layout.

Additionally, if the page has been scrolled then it resets back to the top.

Is it hard to approximate what animated-container does, perhaps explicitly setting the height of the element before and after the transition?
